README.md

```diff
@@ -1,4 +1,143 @@
-# Excelerator
-This is an Excel Addin that reads and writes data to Snowflake
+# Excelerator: Use Snowflake in Excel
+
+Excelerator is an Excel plugin to help you flow data in and out of Snowflake using Excel.
+
+Example use-cases include:
+* financial budgeting write-back,
+* look-up table maintenance,
+* pricing analysis,
+* and more!
+
+| :memo: LEGAL              |
+|:---------------------------|
+|Excelerator is not a supported product by Snowflake or any company. Excelerator will write data to the Snowflake database and should be used with great care. Use at your own risk. |
+
+## Get Started with Excelerator
+
+To get started, you'll need to:
+
+1. Install the plugin in Excel (below)
+2. [Connect your Snowflake account](#use-excelerator)
+3. Then you can [query Snowflake](#execute-a-query) and optionally [write data back](#write-data-to-snowflake)
+
+## Install Excelerator
+
+To begin, clone or download this repository to get your own copy of all the files.
+    
+### Step 1 (optional): Create Stored Procedures in Snowflake
+
+This step is only needed to enable the ‘Data Type Auto-generation’ feature when adding a new table or column. Without the stored procedures, you'll have to define the data types for each new column.
+
+If you have access to the Snowflake web interface and a role with elevated privileges such as "ACCOUNTADMIN", you can run `SnowFlow_Stored_Procedures.sql`.  If you don't have access, then contact your Snowflake DBA and have them run the script for you.  This script will need to be run in the same database that you login into with the Excel Addin, which we discuss below.
+
+### Step 2: Set Required Privileges
+
+Below are the privileges required for each capability.
+
+To **query** data:
+| **Object** | **Privilege**        |
+|------------|----------------------|
+| Database   | USAGE                |
+| Schema     | USAGE, CREATE STAGE* |
+| Table      | SELECT               |
+
+\*Stage can be provided in the login instead
+
+To **upload** data requires everything in query, plus:
+
+| **Object** | **Privilege**            |
+|------------|--------------------------|
+| Schema     | CREATE TABLE             |
+| Table      | INSERT, UPDATE, TRUNCATE |
+
+To **rollback** data requires the schema privileges from upload, plus:
+
+| **Object** | **Privilege** |
+|------------|---------------|
+| Table      | Ownership     |
+  
+    
+For advanced features: Auto-generate Data Types
+
+* usage on all procedures
+* usage on future procedures
+* usage on all functions
+* usage on future functions
+
+One approach for handling these privileges is to create a new role with the above Schema and Database privileges and assigned that role existing role with the Table level privileges.
```

Comment:
We weren't able to figure out what this is suggesting. There _might_ be a typo in this sentence, but we aren't sure what edit it needs.

Reply:
Thanks Adam. Would you like to get together to discuss?

Reply:
Ah yes, I see the typo. I'll fix it and hopefully it will make more sense.